packages/toolkit/src/query/core/buildInitiate.ts

```diff
@@ -45,18 +45,42 @@ type StartQueryActionCreator<
   options?: StartQueryActionCreatorOptions
 ) => ThunkAction<QueryActionCreatorResult<D>, any, any, AnyAction>
 
-export type QueryActionCreatorResult<
+type CommonQueryActionCreatorResult<
   D extends QueryDefinition<any, any, any, any>
-> = Promise<QuerySubState<D>> & {
+> = {
   arg: QueryArgFrom<D>
   requestId: string
   subscriptionOptions: SubscriptionOptions | undefined
   abort(): void
+  unwrap(): Promise<ResultTypeFrom<D>>
   unsubscribe(): void
   refetch(): void
   updateSubscriptionOptions(options: SubscriptionOptions): void
 }
 
+export type QueryActionCreatorResult<
+  D extends QueryDefinition<any, any, any, any>
+> = Promise<QuerySubState<D>> & CommonQueryActionCreatorResult<D>
+
+export type LazyQueryActionCreatorResult<
+  D extends QueryDefinition<any, any, any, any>
+> = Promise<
+  | { data: ResultTypeFrom<D> }
+  | {
+      error:
+        | Exclude<
+            BaseQueryError<
+              D extends QueryDefinition<any, infer BaseQuery, any, any>
+                ? BaseQuery
+                : never
+            >,
+            undefined
+          >
+        | SerializedError
+    }
+> &
+  CommonQueryActionCreatorResult<D>
+
```

Comment:
I'm 99% sure we need to do this, but I'm sure there's a better way. If you [look at this commit](https://github.com/reduxjs/redux-toolkit/pull/1701/commits/fdf4cabd371b1bdc6687e0a159071faf583df6d6), the types for `data` would be wrong as we know for sure in this type of example: `trigger().then((result) => 'data' in result ? result.knownProperty : null)` that `data` would be the Result type, and not `data?: ResultType | undefined`